### PR TITLE
feat: Switch to cryptography library

### LINKED ISSUE #37: Use python native cryptography lib
Python's native cryptography lib is far faster than the python based `ecdsa`, plus it would reduce dependencies.

### PR DESCRIPTION
**Pull request by jrconlin**
closes #37

### CODE REVIEW

File: python/py_vapid/jwt.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    
+    :return tuple containing the signature material and signature
+    
+    """
+    payload, asig = auth.encode('utf8').rsplit(b'.', 1)
+    sig = b64urldecode(asig)
+    if len(sig) != 64:
+        return payload, sig
+
+    encoded = utils.encode_dss_signature(
```
Comment (kitcambridge) [inline python/py_vapid/jwt.py]: Oh, you were telling me about this! Just to make sure I understand, this is to decode OpenSSL signatures, and the `len(sig) != 64` branch above is to decode ones signed with this library?

File: python/py_vapid/__init__.py [hunk 2]
```diff
@@ -100,54 +98,69 @@ def from_file(cls, private_key_file=None):
...
+        if value:
+            self._public_key = self.private_key.public_key()
 
     @property
     def public_key(self):
         """The VAPID public ECDSA key
 
         The public key is currently read only. Set it via the `.private_key`
-        method.
+        method. This will autogenerate a public and private key if no value
```
Comment (jrconlin) [inline python/py_vapid/__init__.py]: Yep, I realized that doing the "optimization" could break some installs, forgot to clean this one up.

File: python/py_vapid/__init__.py [hunk 3]
```diff
@@ -52,41 +43,48 @@ def __init__(self, private_key=None):
...
         """Initialize VAPID using a private key in PEM format.
 
         :param private_key: A private key in PEM format.
-        :type private_key: str
+        :type private_key: bytes
 
         """
-        key = ecdsa.SigningKey.from_pem(private_key)
-        return cls(key)
+        # not sure why, but load_pem_private_key fails to deserialize
```
Comment (jrconlin) [inline python/py_vapid/__init__.py]: It's in the test: https://github.com/web-push-libs/vapid/pull/42/files#diff-f4037d7f16c50099d22585d0d3b77b90R30

I'm constructing the PEM from the b64 encoded DER, which is pretty much what a PEM is. That's why I can strip the header/footer and then concat the strings to decode into a DER.

File: python/py_vapid/jwt.py (hunk 1 above)
Comment (jrconlin) [inline python/py_vapid/jwt.py]: Yeah, basically, an ecdsa lib encodes to a serialized int pair, which is 64 bytes. OpenSSL returns a full DER. Unfortunately, OpenSSL wants a DER sequence when reading values back in. 

I really ought to also check for a raw key sequence DER ("\04" + 64) and convert that. Thanks!

File: python/py_vapid/__init__.py [hunk 4]
```diff
@@ -52,41 +43,48 @@ def __init__(self, private_key=None):
 
         """
         self.private_key = private_key
+        if private_key:
```
Comment (jrconlin) [inline python/py_vapid/__init__.py]: Ah, I won't call `_private key`, but I will call `private_key`, this will raise a useful VapidException.

**Pull request merged**